app/models/policy_host.rb

```diff
@@ -9,8 +9,19 @@ class PolicyHost < ApplicationRecord
   validates :host, presence: true, on: :create
   validates :host_id, presence: true, uniqueness: { scope: :policy }
 
+  scope :supported_host_ids, lambda { |policy_id, host_ids|
+    profile = Profile.find_by(policy_id: policy_id)
+    host_ids.map do |host_id|
+      host = Host.find_by(id: host_id)
+      if profile.supported_os_versions.include?(Gem::Version.new("#{host&.os_major_version}.#{host&.os_minor_version}"))
+        host_id
+      end
+    end.compact
+  }
+
   def self.import_from_policy(policy_id, host_ids)
-    import(host_ids.map do |host_id|
+    supported_host_ids = supported_host_ids(policy_id, host_ids)
```

Comment:
We should not silently ignore, but fail if something is not supported. Something like:
```
validates :host, -> { supported_host_ids.include?(os_minor_version) }
```

Reply:
Good idea, this is kind of what I had in mind too for the validation.